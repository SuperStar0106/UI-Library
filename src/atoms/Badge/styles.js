import styled from 'styled-components'

import { getVariantColor } from '../../common/styles/form'
import helpers from '../../theme/helpers'

const { colors, fontSize, padding, radius, textStyles } = helpers

export const Badge = styled.div`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  min-width: ${props => props.size};
  max-width: ${props => (props.length === 1 ? props.size : '100%')};
  height: ${props => props.size};
  padding-right: ${props =>
    props.length === 1 ? 0 : padding(props.padding)(props) || props.padding || null};
  padding-left: ${props =>
    props.length === 1 ? 0 : padding(props.padding)(props) || props.padding || null};
  color: ${colors('white')};
  ${textStyles('badge')};
  font-size: ${props => fontSize(props.fontsize)(props) || props.fontsize || null};
  background-color: ${props => getVariantColor(props.variant)};
  border-radius: ${props => radius(props.radius)(props) || props.radius || props.size};
`

export default Badge
