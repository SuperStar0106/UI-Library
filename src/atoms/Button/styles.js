import styled, { css, keyframes } from 'styled-components'
import { bool, string, oneOf } from 'prop-types'
import Link from 'react-router-dom/Link'
import { media, overflowEllipsis } from '../../utils/utils'
import {
  boxShadow,
  color,
  rgba,
  gutter,
  fontSize,
  fontWeight,
  fontFamily,
  letterSpacing,
  radius,
  roundedButtonSize,
  textStyles,
  transition
} from '../../utils/theme'

const pulsing = keyframes`
  from {
    transform: scale(1);
    opacity: .1;
  }
  to {
    transform: scale(1.8);
    opacity: .2;
  }
`

export const slidingBackgroundStyles = css`
  &::before {
    content: ' ';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${rgba('white', 0.15)};
    transform: scaleX(0);
    transform-origin: 0;
    transition: ${transition('md')};
  }

  &:hover::before {
    transform: scaleX(1);
  }
`

export const sizeMdStyles = css`
  padding: ${gutter('md')} ${gutter('lg')};
  ${textStyles('button')};

  ${media.mobile`
  padding: ${gutter('md')};
  `};
`

export const sizeXsStyles = css`
  padding: ${gutter('xs')};
  ${textStyles('button_small')};
`

export const sizeSmStyles = css`
  padding: ${gutter('xs')} ${gutter('sm')};
  ${textStyles('button_small')};
`

export const sizeLgStyles = css`
  padding: ${gutter('md')} ${gutter('xl')};
  font-size: ${fontSize('md')};
  font-weight: ${fontWeight('bold')};
  letter-spacing: ${letterSpacing('lg')};
`

export const primaryStyles = css`
  background: ${color('seafoamblue')};
  color: ${color('white')};

  &:hover {
    color: ${color('white')};
  }

  &[disabled] {
    color: ${color('white')};
    background-color: ${color('gray', '200')};
  }
`

export const primaryOutlineStyles = css`
  background-color: ${color('white')};
  color: ${color('seafoamblue')};
  border: 1px solid ${color('seafoamblue')};
  font-weight: ${fontWeight('bold')};

  &:hover {
    background-color: ${color('green', '100')};
    color: ${color('seafoamblue')};
  }
`

export const closeDarkStyles = css`
  background: ${rgba('black', '0.3')};
  color: ${color('white')};

  &:hover {
    color: ${color('white')};
  }

  &::before {
    background: ${rgba('black', 0.15)};
  }

  svg path {
    fill: ${color('white')};
  }
`

export const darkStyles = css`
  background: ${color('gunmetal')};
  color: ${color('white')};

  &:hover {
    color: ${color('white')};
  }

  &::before {
    background: ${rgba('white', 0.05)};
  }

  svg path {
    fill: ${color('white')};
  }
`

export const lightStyles = css`
  background: ${color('white')};
  color: ${color('texts', 'light')};

  &:hover {
    color: ${color('texts', 'light')};
  }

  &::before {
    background: ${rgba('black', 0.02)};
  }

  svg path {
    fill: ${color('texts', 'xlight')};
  }
`

export const neutralStyles = css`
  background: ${color('gray', '150')};
  color: ${color('gray', '300')};

  &:hover {
    color: ${color('gray', '300')};
  }

  &::before {
    background: ${rgba('black', 0.02)};
  }

  svg path {
    fill: ${color('gray', '300')};
  }
`

export const dangerStyles = css`
  background: ${color('white')};
  color: ${color('alabamacrimson')};

  &:hover {
    color: ${color('alabamacrimson')};
  }

  &::before {
    background: ${rgba('black', 0.02)};
  }

  svg path {
    fill: ${color('alabamacrimson')};
  }
`

export const dangerReverseStyles = css`
  background: ${color('alabamacrimson')};
  color: ${color('white')};

  &:hover {
    color: ${color('white')};
  }

  &::before {
    background: ${rgba('black', 0.02)};
  }

  svg path {
    fill: ${color('white')};
  }
`

export const warningStyles = css`
  background: ${color('white')};
  color: ${color('orange', '400')};

  &:hover {
    color: ${color('orange', '400')};
  }

  &::before {
    background: ${rgba('black', 0.02)};
  }

  svg path {
    fill: ${color('orange', '400')};
  }
`

export const nakedStyles = css`
  padding: 0;
  color: ${color('texts', 'light')};
  box-shadow: none;
  border: none;
  font-size: ${fontSize('md')};
  letter-spacing: ${letterSpacing('md')};

  &:hover {
    color: ${color('texts', 'dark')};
  }

  svg path {
    fill: ${color('texts', 'xlight')};
  }
`

export const linkedinStyles = css`
  background: ${color('linkedin', 'secondary')};
  color: ${color('white')};

  &[disabled] {
    background: ${color('grey', '200')};
  }

  &:hover {
    color: ${color('white')};
  }
`

export const roundedStyles = css`
  padding: 0;
  align-items: center;
  justify-content: center;
  transform: translateZ(0);
  border-radius: 50%;

  svg {
    display: block;
    transform: rotate(0deg);
    backface-visibility: hidden;
    -webkit-transform-style: preserve-3d;
    -webkit-transform: translate3D(0, 0, 0) scale(1) skewY(0deg);
  }
`

export const roundedSizeAutoStyles = css`
  width: auto;
  height: auto;
`

export const roundedSizeSmStyles = css`
  width: ${roundedButtonSize('sm')};
  height: ${roundedButtonSize('sm')};
`

export const roundedSizeMdStyles = css`
  width: ${roundedButtonSize('md')};
  height: ${roundedButtonSize('md')};
`

export const roundedSizeLgStyles = css`
  width: ${roundedButtonSize('lg')};
  height: ${roundedButtonSize('lg')};
`

export const borderRoundedStyles = css`
  border-radius: 40px;
`

export const fullWidthStyles = css`
  width: 100%;
  &:not(:last-child) {
    margin-bottom: ${gutter('lg')};
  }
`

export const pulsingStyles = css`
  overflow: visible;

  &::before {
    position: absolute;
    width: 100%;
    height: 100%;
    content: ' ';
    background: inherit;
    opacity: 0.1;
    border-radius: 50%;
    transform-origin: center;
    animation: ${pulsing} 1.5s ease infinite alternate;
  }
`

export const ButtonText = styled.span``

export const ButtonLabel = styled.span`
  display: inline-block;
`

export const withIconStyles = css`
  text-align: left;
`

function getButtonMode(mode) {
  switch (mode) {
    case 'primary':
      return primaryStyles
    case 'primary-outline':
      return primaryOutlineStyles
    case 'success':
      return primaryStyles
    case 'close-dark':
      return closeDarkStyles
    case 'dark':
      return darkStyles
    case 'light':
      return lightStyles
    case 'neutral':
      return neutralStyles
    case 'danger':
      return dangerStyles
    case 'danger-reverse':
      return dangerReverseStyles
    case 'error':
      return dangerStyles
    case 'warning':
      return warningStyles
    case 'naked':
      return nakedStyles
    case 'linkedin':
      return linkedinStyles
    default:
      return primaryStyles
  }
}

function getButtonSize(size, rounded) {
  if (rounded) {
    switch (size) {
      case 'auto':
        return roundedSizeAutoStyles
      case 'sm':
        return roundedSizeSmStyles
      case 'md':
        return roundedSizeMdStyles
      case 'lg':
        return roundedSizeLgStyles
      default:
        return roundedSizeMdStyles
    }
  } else {
    switch (size) {
      case 'xs':
        return sizeXsStyles
      case 'sm':
        return sizeSmStyles
      case 'md':
        return sizeMdStyles
      case 'lg':
        return sizeLgStyles
      default:
        return sizeMdStyles
    }
  }
}

export const inlineLinkStyles = css`
  display: inline-block;
  position: relative;
  color: ${color('texts', 'dark')};
  font-weight: ${fontWeight('regular')};
  line-height: 1.8;

  &::after {
    position: absolute;
    bottom: 2px;
    left: 0;
    width: 100%;
    height: 1px;
    background-color: ${color('gray', '200')};
    transition: ${transition('md')};
    content: ' ';
  }

  &:hover {
    &::after {
      background-color: ${color('seafoamblue')};
    }
  }
`

export const buttonStyles = css`
  position: relative;
  display: inline-flex;
  align-items: center;
  align-self: ${({ alignself }) => alignself || null};
  justify-content: ${({ align }) => align || 'center'};
  width: auto;
  font-family: ${fontFamily('texts')};
  text-align: ${({ align }) => align || 'center'};
  text-transform: uppercase;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
  border: none;
  box-shadow: ${({ shadow }) => shadow || boxShadow('buttons')};
  appearance: none;
  overflow: hidden;
  border-radius: ${radius('sm')};
  transition: ${transition('sm')};
  line-height: 1;

  ${slidingBackgroundStyles};

  &:active {
    transform: translateY(2px);
    box-shadow: 0 0 4px ${rgba('black', 0.2)};
  }

  &:hover {
    ${media.mobile`
      transform: translateY(2px);
    `}
  }

  &[disabled] {
    pointer-events: none;
  }

  svg {
    display: block;

    & + span,
    & + ${ButtonText} {
      margin-left: ${gutter('sm')};
    }
  }

  ${ButtonText},
  span {
    ${overflowEllipsis};
    & + svg {
      margin-left: ${gutter('sm')};
    }
  }

  ${props => getButtonSize(props.size, props.rounded)}
  ${props => getButtonMode(props.mode)}
  ${props => (props.rounded ? roundedStyles : null)}
  ${props => (props.borderRounded ? borderRoundedStyles : null)}
  ${props => (props.pulsing ? pulsingStyles : null)}
  ${props => (props.withicon ? withIconStyles : null)}
  ${props => (props.width === 'full' ? fullWidthStyles : null)}
`

/** @component */
export const Button = styled.button`
  ${buttonStyles};
  ${ButtonLabel} {
    margin-left: ${props =>
      props.withicon ? props.theme.buttonIconWidth : props.theme.buttonIconWidth};
  }
`

export const LinkButton = styled(Link)`
  ${buttonStyles};
  ${ButtonLabel} {
    margin-left: ${props =>
      props.withicon ? props.theme.buttonIconWidth : props.theme.buttonIconWidth};
  }
`

LinkButton.propTypes = {
  to: string.isRequired,
  ...Button.propTypes
}

export const HrefButton = styled.a`
  ${buttonStyles};
  ${ButtonLabel} {
    margin-left: ${props =>
      props.withicon ? props.theme.buttonIconWidth : props.theme.buttonIconWidth};
  }
`

HrefButton.propTypes = {
  href: string.isRequired,
  ...Button.propTypes
}

export const InlineLink = styled(Link)`
  ${inlineLinkStyles};
`

InlineLink.propTypes = {
  to: string.isRequired,
  ...Button.propTypes
}

export const InlineHref = styled.a`
  ${inlineLinkStyles};
`

InlineHref.propTypes = {
  to: string.isRequired,
  ...Button.propTypes
}

export const ButtonIcon = styled.span`
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  padding: ${gutter('sm')};
  background: ${rgba('black', 0.1)};
`

export const ButtonInlineIcon = styled.span`
  display: flex;
  align-items: center;

  & + ${ButtonText}, & + span {
    margin-left: ${gutter('xs')};
  }
`

export default Button

export const ButtonsGroup = styled.div`
  display: flex;
  justify-content: center;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: ${gutter('lg')};
  max-width: ${props => (props.maxwidth ? props.maxwidth : '100%')};

  &:last-child {
    margin-bottom: 0;
  }

  button,
  a {
    flex: 1 1 0;
    margin: 0 ${gutter('sm')};

    &:first-child {
      margin-left: 0;
    }

    &:last-child {
      margin-right: 0;
    }

    ${media.mobile`
      margin: 0 ${gutter('xs')};
    `};
  }
`
