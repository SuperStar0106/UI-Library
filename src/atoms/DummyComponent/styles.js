import styled from 'styled-components'

import { get } from '../../theme/helpers'

export const DummyComponent = styled.div`
  position: relative;
  color: ${get('color', 'dark')};
  background: ${get('color', 'white')};
`

export default DummyComponent
