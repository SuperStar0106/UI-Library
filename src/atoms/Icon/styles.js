import styled, { css } from 'styled-components'

import { color } from '../../utils/theme'

const iconSvgStrokedStyles = css`
  g, path {
    stroke: inherit;
    fill: none;
  }
`

const iconSvgFilledStyles = css`
  g, path {
    fill: inherit;
    stroke: none;
  }
`

export const IconSvg = styled.svg`
  ${({ stroked }) => stroked ? iconSvgStrokedStyles : iconSvgFilledStyles};
`