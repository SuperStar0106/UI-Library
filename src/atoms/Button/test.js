import React from 'react'

import { render } from '../../utils/tests'

import { Button, LinkButton, HrefButton } from './styles'

test.skip('<Button> renders correctly and has white text by default', () => {
  const button = render(<Button>Test button</Button>).toJSON()
  expect(button).toMatchSnapshot()
  expect(button.children).toContain('Test button')
  expect(button).toHaveStyleRule('color', '#FFFFFF')
  expect(button).toHaveStyleRule('background', '#00C29A')
})

test.skip('<Button> has correct colour for mode light', () => {
  const button = render(<Button mode="light">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#C3C3C6')
  expect(button).toHaveStyleRule('background', '#FFFFFF')
})

test.skip('<Button> has correct colour for mode dark', () => {
  const button = render(<Button mode="dark">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#FFFFFF')
  expect(button).toHaveStyleRule('background', '#2C2D34')
})

test.skip('<Button> has correct colour for mode neutral', () => {
  const button = render(<Button mode="neutral">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#373942')
  expect(button).toHaveStyleRule('background', '#D7D7D9')
})

test.skip('<Button> has correct colour for mode danger', () => {
  const button = render(<Button mode="danger">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#F35454')
  expect(button).toHaveStyleRule('background', '#FFFFFF')
})

test.skip('<Button> has correct colour for mode linkedin', () => {
  const button = render(<Button mode="linkedin">Test button</Button>).toJSON()
  expect(button).toHaveStyleRule('color', '#FFFFFF')
  expect(button).toHaveStyleRule('background', '#0077B5')
})

test.skip('<LinkButton> renders correctly', () => {
  const button = render(<LinkButton to="#nowhere">Link button</LinkButton>).toJSON()
  expect(button.children).toContain('Link button')
  expect(button).toHaveStyleRule('color', '#FFFFFF')
})

test.skip('<HrefButton> renders correctly', () => {
  const button = render(<HrefButton href="#nowhere">Href button</HrefButton>).toJSON()
  expect(button.children).toContain('Href button')
  expect(button).toHaveStyleRule('color', '#FFFFFF')
})
