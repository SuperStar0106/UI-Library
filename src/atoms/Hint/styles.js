import styled, { css } from 'styled-components'

import { getVariantColor } from '../../common/styles/form'
import { get } from '../../theme/helpers'

export const Hint = styled.div(
  props => css`
    font-family: ${get('fontFamily', 'texts')};
    color: ${getVariantColor(props.variant, get('color', 'mute', 'default'))};
    ${get('text', 'hint')};
    margin-top: ${get('gutter', 'xs')};
  `
)

export default Hint
