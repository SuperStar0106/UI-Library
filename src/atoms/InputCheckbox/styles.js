import styled, { css } from 'styled-components'

// common form styles
import { fieldTypeStyles } from '../../common/styles/form'

import { get } from '../../theme/helpers'

const checkedStyles = () => css`
  background-color: ${get('color', 'primary')};
  border-color: ${get('color', 'primary')};
  &::after {
    opacity: 1;
  }
`

export const InputCheckbox = styled.div`
  ${fieldTypeStyles};
  position: relative;
  width: 15px;
  height: 15px;
  padding: 0;
  cursor: pointer;
  order: ${props => props.order || null};
  border-radius: ${get('radius', 'sm')};
  transition: ${get('transition', 'sm')};

  &::after {
    content: 'âœ“';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    line-height: 1;
    text-align: center;
    color: white;
    opacity: 0;
    transition: ${get('transition', 'sm')};
  }

  ${props => props.checked && checkedStyles()}
`

export default InputCheckbox
