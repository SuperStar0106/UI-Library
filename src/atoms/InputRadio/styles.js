import styled from 'styled-components'

import { get } from '../../theme/helpers'
import { overflowEllipsis } from '../../common/styles/text'

export const InputRadioWrapper = styled.div`
  flex: 1;
`

export const InputRadio = styled.input`
  position: absolute;
  top: 0;
  left: 0;
  opactiy: 0;
`

export const InputRadioLabel = styled.label`
  position: relative;
  cursor: pointer;
  user-select: none;
  display: block;
  padding: ${get('gutter', 'xs')};
  padding-right: calc(${get('gutter', 'xs')} + ${get('borderWidth', 'input')});
  text-align: center;
  background-color: ${get('color', 'bg', 'light')};
  border: ${get('borderWidth', 'input')} solid ${get('color', 'border', 'primary')};
  margin-left: -${get('borderWidth', 'input')};
  ${get('textStyles', 'input')};
  transition: ${get('transition', 'sm')};
  line-height: 1;
  ${overflowEllipsis};

  ${InputRadio}:checked + & {
    z-index: 2;
  }
  &:hover,
  ${InputRadio}:checked + & {
    background-color: ${get('color', 'white')};
    box-shadow: ${get('boxShadow', 'xs')};
  }

  ${InputRadio}:checked + & {
    border-color: ${get('color', 'primary')};
    font-weight: ${get('fontWeight', 'medium')};
  }

  ${InputRadioWrapper}:first-child & {
    border-radius: ${get('radius', 'md')} 0 0 ${get('radius', 'md')};
    margin-left: 0;
  }

  ${InputRadioWrapper}:last-child & {
    border-radius: 0 ${get('radius', 'md')} ${get('radius', 'md')} 0;
    padding-right: ${get('gutter', 'xs')};
  }
`
