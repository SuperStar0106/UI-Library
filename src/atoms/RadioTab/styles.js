import styled, { css } from 'styled-components'

import { get, getCss } from '../../theme/helpers'
import { overflowEllipsis } from '../../common/styles/text'

//molecules
import { Label } from '../../molecules/Label/styles'

export const StyledRadioTab = styled.div`
  flex: 1;
`

export const Input = styled.input`
  position: absolute;
  top: 0;
  left: 0;
  opacity: 0;
`

const checkedStyles = css`
  color: ${get('color', 'primary', 'dark')};
  border-color: ${get('color', 'primary', 'default')};
  background-color: ${get('color', 'light', 'light')};
  box-shadow: ${get('boxShadow', 'xs')};
  z-index: 2;
`

const columnStyles = css`
  ${Label} {
    margin-top: -${get('borderWidth', 'input')};
  }

  &:first-child ${Label} {
    border-radius: ${get('radius', 'md')} ${get('radius', 'md')} 0 0;
    margin-top: 0;
  }

  &:last-child ${Label} {
    border-radius: 0 0 ${get('radius', 'md')} ${get('radius', 'md')};
  }
`

const rowStyles = css`
  ${Label} {
    margin-left: -${get('borderWidth', 'input')};
  }

  &:first-child ${Label} {
    border-radius: ${get('radius', 'md')} 0 0 ${get('radius', 'md')};
    margin-left: 0;
  }

  &:last-child ${Label} {
    border-radius: 0 ${get('radius', 'md')} ${get('radius', 'md')} 0;
  }
`

export const radioTabStyles = props => css`
  flex: 1;
  ${Label} {
    ${getCss('fields', 'default')};
    flex-grow: 1;
    position: relative;
    display: block;
    max-width: 100%;
    user-select: none;
    margin: 0;
    padding: ${get('gutter', 'xs')};
    padding-right: calc(${get('gutter', 'xs')} + ${get('borderWidth', 'input')});
    cursor: pointer;
    line-height: 1;
    text-align: center;
    transition: ${get('transition', 'sm')};
    border-radius: 0;
    ${overflowEllipsis};

    ${props.checked && checkedStyles};

    &:hover {
      box-shadow: ${get('boxShadow', 'xs')};
    }
  }

  ${props.direction === 'column' && columnStyles};
  ${props.direction === 'row' && rowStyles};
`
