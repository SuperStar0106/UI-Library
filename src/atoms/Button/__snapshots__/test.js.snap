// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly and has white text by default 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  font-family: Arial;
  text-align: center;
  text-transform: uppercase;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  border-radius: 3px;
  -webkit-transition: all .2s cubic-bezier(0.41,0.094,0.54,0.07);
  transition: all .2s cubic-bezier(0.41,0.094,0.54,0.07);
  line-height: 1;
  padding: 0.32rem 0.48rem;
  font-size: 0.22rem;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  background: #00C29A;
  color: #FFFFFF;
}

.c0::before {
  content: ' ';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255,255,255,0.15);
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transform-origin: 0;
  -ms-transform-origin: 0;
  transform-origin: 0;
  -webkit-transition: all .3s cubic-bezier(0.41,0.094,0.54,0.07);
  transition: all .3s cubic-bezier(0.41,0.094,0.54,0.07);
}

.c0:hover::before {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c0:active {
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
  box-shadow: 0 0 4px rgba(undefined,0.2);
}

.c0[disabled] {
  pointer-events: none;
}

.c0 svg {
  display: block;
}

.c0 svg + span,
.c0 svg + .c1 {
  margin-left: 0.26rem;
}

.c0 .c1,
.c0 span {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 .c1 + svg,
.c0 span + svg {
  margin-left: 0.26rem;
}

.c0:hover {
  color: #FFFFFF;
}

.c0[disabled] {
  color: #FFFFFF;
}

.c0 .c2 {
  margin-left: 0.92rem;
}

@media (max-width:670px) {
  .c0:hover {
    -webkit-transform: translateY(2px);
    -ms-transform: translateY(2px);
    transform: translateY(2px);
  }
}

@media (max-width:670px) {
  .c0 {
    padding: 0.32rem;
  }
}

<button
  className="c0"
>
  Test button
</button>
`;
